### PR TITLE
Fix linear solvers for PDALs

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Re-enables the linear solver tests that were deferred from #1535.

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]
Looks fine to me given that it compiles and passes tests :)

**Comment by kidder** [comment]
@nilsleiffischer this isn't compiling...

**Comment by nilsvu** [comment]
@kidder @nilsdeppe This had an issue with a GCC warning on Travis, which occurred only for GCC6 and only in Release mode. I could reproduce it locally, but I have no idea what GCC6's problem with the code is. Since more recent versions of the compiler don't complain, I silence the warning in the last commit.

This PR is going through Travis right now, but very slowly. Could you review so that we can merge when Travis has decided it is happy?

**Review by kidder — DISMISSED** [review]

File: src/DataStructures/DataBox/DataBox.hpp [hunk 1]
```diff
@@ -925,6 +925,13 @@ constexpr void DataBox<tmpl::list<Tags...>>::merge_old_box(
        '0')...};
 }
 
+// Silence "maybe-uninitialized" warning for GCC-6. The warning only occurs in
+// Release mode.
+// Note that clang also defines `__GNUC__`, so we need to exclude it.
```
Comment (nilsdeppe) [inline src/DataStructures/DataBox/DataBox.hpp]: can this warning pop really not be scoped more than this?

Comment (nilsvu) [reply src/DataStructures/DataBox/DataBox.hpp]: Do you mean move it out of `DataBox.hpp`? I can try..

Comment (nilsvu) [reply src/DataStructures/DataBox/DataBox.hpp]: I moved it to the two linear solver tests that emit the warning. It didn't work to just wrap the problematic function call in the pragmas, but I had to also wrap the include of `DataBox.hpp`. Do you like this approach better than wrapping the problematic constructor in `DataBox.hpp`?

What I really don't understand is why the issue occurs in `CG/Test_ElementActions.cpp` and `GMRES/Test_ResidualMonitorActions.cpp`, but _not_ in `GMRES/Test_ElementActions.cpp` and `CG/Test_ResidualMonitorActions.cpp`, which are basically the same...

Comment (nilsdeppe) [reply src/DataStructures/DataBox/DataBox.hpp]: Sorry, I meant putting it around the one line that causes the error, not the whole constructor. Is that not possible?

Comment (nilsvu) [reply src/DataStructures/DataBox/DataBox.hpp]: ok that works

**Comment by nilsvu** [comment]
@nilsdeppe @kidder Travis is finally happy with this. Could you have another look and merge?

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsvu** [comment]
Thanks! The post-pdals storyline continues in #1575 :)

**Pull request merged**